micro-rdk/src/common/conn/server.rs

```diff
@@ -356,9 +354,18 @@ where
             network,
         }
     }
-    pub async fn serve(&mut self, robot: Arc<Mutex<LocalRobot>>) {
+    pub async fn serve(&mut self, robot: Arc<Mutex<LocalRobot>>, app_client: Option<AppClient>) {
         let cloned_robot = robot.clone();
 
+        if let Some(app_client) = app_client {
+            self.app_client.write().await.replace(app_client);
+        }
+
+        // Let the robot register any periodic app client tasks it may
+        // have based on its configuration.
+        self.app_client_tasks
+            .append(&mut robot.lock().unwrap().get_periodic_app_client_tasks());
```

Comment:
So is the idea to load feature-gated tasks within this function in `LocalRobot`, with user-defined tasks loaded through the entrypoint?

Reply:
Nothing so organized was intended. I was just doing what I had to do to get a robot running data collection before the network was able to access app.viam.

Reply:
@mattjperez - OK to resolve?